frontend/src/components/details/TableComponent.jsx

```diff
@@ -3,16 +3,15 @@ import React, { Component } from 'react';
 import { Container } from 'react-grid-system';
 import {
     getImpactCategoriesTableHeaders,
-    getImpactCategoriesTableData
+    getImpactAssessmentData
 } from 'interface/projectInterface';
-
 /**
  * displays the impact catagories table of the selected model of the related product.
  */
 class TableComponent extends Component {
     state = {
-        headers: getImpactCategoriesTableHeaders(this.props.modelId),
-        rows: getImpactCategoriesTableData(this.props.modelId)
+        headers: getImpactCategoriesTableHeaders(),
+        rows: getImpactAssessmentData()
```

Comment:
how do these know, what data to fetch without passing any data to them here?

Reply:
For now, we are only getting the calculation result based on the Project Id and not the model Id. We need to implement the calculation call based on model Id at a later point in time. Also, the calculation data with respect to model Id will also be called from BackendConnect.js processed at ProjectInterface.js and processBackendData.js. So don't think we need to pass any value here...we only send one API request and process the calculation data for filling the details page.